Close #18080 Update api_products.class.php for MULTIPRICES support

matthieu-michou-wattandsea opened the pull request:
PRODUIT_MULTIPRICES is currently supported by get function, but not post and put
this code, credited to @ruemmlernet,  adds PRODUIT_MULTIPRICES support in post and put functions

# FIX|Fix #18080
allow to pass objects with multiprices during creation or update of a product

example with python Dolibarr API  and 7 levels of prices: 
```
p = {   
    "ref": " P9999",
    "label": "test product", 
    "price_base_type": "HT",
    "default_vat_code": "20",
    "tva_tx": "20.0000",
    "price": 199.98,
    "multiprices": {"1": "199.98", "2": "180", "3": "170", "4": "160", "5": "150", "6": "140", "7": "139.99"},
    "multiprices_base_type": {"1": "HT", "2": "HT", "3": "HT", "4": "HT", "5": "HT", "6": "HT", "7": "HT"},
    "multiprices_tva_tx": {"1": "20", "2": "20", "3": "20", "4": "20", "5": "20", "6": "20", "7": "20"}
}
last_id = dolibarr_inst.call_create_api('products', p)
print( dolibarr_inst.call_get_api('products', last_id) )
```

matthieu-michou-wattandsea committed:
Update api_products.class.php

Update api_products.class.php

matthieu-michou-wattandsea committed:
Merge branch 'develop' into branche-developpement-MM

matthieu-michou-wattandsea commented:
Hello @eldy  
This is one of my first pull-requests, I hope this is clean enough as I don't want to make mess in the project 
This original fix has been proposed in 2021 but never merged in main project, however it works fine as it and is useful as well : I currently use it for mass import from another ERP in a V18
I made slight modifications to use `getDolGlobalString` e.g : 
`if ($conf->global->PRODUIT_MULTIPRICES)` to `if (getDolGlobalString('PRODUIT_MULTIPRICES'))`

Let me know can I help further to make it integrate in the dev version
thanks

eldy commented:
Your PR looks great. 
Thanks.
However it is for the moment postponed because we don't include new feature support until the branch 19.0 has been created. I will merge it in develop just after 19.0 branch creation.

matthieu-michou-wattandsea committed:
Update api_products.class.php

Prevent a php error if the quantity is 0 or negative

matthieu-michou-wattandsea committed:
API REST : allow to set date_creation part#1

this commit manages the field date_creation if used in the API contacts / thirparties / products

matthieu-michou-wattandsea committed:
API REST : allow to set date_creation part#1

this commit manages the field date_creation if used in the API contacts / thirparties / products

matthieu-michou-wattandsea committed:
Use of custom products labels in main views

Allow to edit products labels in main views

matthieu-michou-wattandsea committed:
Revert "Use of custom products labels in main views"

This reverts commit c7844cf8fdde7b50f8085ad6b9f8fca8f15bc72d.

matthieu-michou-wattandsea committed:
Merge branch 'branche-developpement-MM' of https://github.com/matthieu-michou-wattandsea/dolibarr into branche-developpement-MM

matthieu-michou-wattandsea committed:
Revert "Revert "Use of custom products labels in main views""

This reverts commit 62589bf38925f0083b79f0483462bcac427e4a5b.

matthieu-michou-wattandsea committed:
Revert "Use of custom products labels in main views"

This reverts commit c7844cf8fdde7b50f8085ad6b9f8fca8f15bc72d.

matthieu-michou-wattandsea committed:
Revert "Revert "Use of custom products labels in main views""

This reverts commit 62589bf38925f0083b79f0483462bcac427e4a5b.

matthieu-michou-wattandsea committed:
typos
